Engine/src/Engine/InputSystem.cpp

```diff
@@ -0,0 +1,113 @@
+#include "Engine/InputSystem.h"
+
+using namespace Engine;
+
+InputSystem* InputSystem::m_instance = nullptr;
+std::mutex InputSystem::m_instanceMutex;
+
+InputSystem::InputSystem():m_window(nullptr), m_isInit(false)
+{
+	//Moved map initialization to the init function

```

Comment:
Since this was moved might want to just remove this constructor definition

Reply:
Constructor was made private to conform to the singleton pattern.